icu4c/source/common/brkiter.cpp

```diff
@@ -438,7 +438,9 @@ BreakIterator::makeInstance(const Locale& loc, int32_t kind, UErrorCode& status)
             if (U_SUCCESS(kvStatus) && (value == "strict" || value == "normal" || value == "loose")) {
                 uprv_strcat(lbType, "_");
                 uprv_strcat(lbType, value.data());
-            } else {
+            }
```

Comment:
The variable lbType (the char array on the stack is not used except inside the case UBRK_LINE block, please move it inside the UBRK_LINE block and rename it  (from lbType to type? or lb_lw?) since now it is not just lb.